inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -3240,6 +3240,20 @@ private <T> BeanDefinition<T> lastChanceResolve(Argument<T> beanType,
             return bean;
         }
 
+        // try resolve @DefaultImplementation
+        for (BeanDefinition<T> candidate : candidates) {
+            if (candidate.hasStereotype(DefaultImplementation.class)) {
```

Comment:
Checking the first candidate should be enough to know if there is a default impl